openpype/hosts/maya/plugins/inventory/connect_geometry.py

```diff
@@ -0,0 +1,139 @@
+
+        # Validate to only 1 source container.
+        source_containers = containers_by_family.get("animation", [])
+        source_containers += containers_by_family.get("pointcache", [])
+        source_container_namespaces = [
+            x["namespace"] for x in source_containers
+        ]
+        message = (
+            "{} animation containers selected:\n\n{}\n\nOnly select 1 of type "
+            "\"animation\" or \"pointcache\".".format(
+                len(source_containers), source_container_namespaces
+            )
+        )
+        if len(source_containers) != 1:
+            self.display_warning(message)
+            return
+
+        source_object = source_containers[0]["objectName"]
+
+        # Collect matching geometry transforms based cbId attribute.
+        target_containers = []
+        for family, containers in containers_by_family.items():
+            if family in ["animation", "pointcache"]:
+                continue
+
+            target_containers.extend(containers)
+
+        source_data = self.get_container_data(source_object)
+        matches = []
+        node_types = []
+        for target_container in target_containers:
+            target_data = self.get_container_data(
+                target_container["objectName"]
+            )
+            node_types.extend(target_data["node_types"])
+            for id, transform in target_data["ids"].items():
+                source_match = source_data["ids"].get(id)
+                if source_match:
+                    matches.append([source_match, transform])
+
+        # Message user about what is about to happen.
+        if not matches:
+            self.display_warning("No matching geometries found.")
+            return
+
+        message = "Connecting geometries:\n\n"
+        for match in matches:
+            message += "{} > {}\n".format(match[0], match[1])
+
+        choice = self.display_warning(message, show_cancel=True)
+        if choice is False:
+            return
+
+        # Setup live worldspace blendshape connection.
+        for match in matches:
+            source = match[0]
+            target = match[1]
+            blendshape = cmds.blendShape(source, target)[0]
+            cmds.setAttr(blendshape + ".origin", 0)
+            cmds.setAttr(blendshape + "." + target.split(":")[-1], 1)
+
+        # Update Xgen if in any of the containers.
+        if "xgmPalette" in node_types:
+            cmds.xgmPreview()
+
+    def get_container_data(self, container):
+        data = {"node_types": [], "ids": {}}
+        ref_node = cmds.sets(container, query=True, nodesOnly=True)[0]
+        for node in cmds.referenceQuery(ref_node, nodes=True):
+            node_type = cmds.nodeType(node)
+            data["node_types"].append(node_type)
+            if node_type == "mesh":
+                transform = cmds.listRelatives(node, parent=True)[0]
+                id = cmds.getAttr(transform + ".cbId")
+                data["ids"][id] = transform
```

Comment:
Cant rely on shape ids because they can change upstream due to deformation, but the transform id should match.

Reply:
> Cant rely on shape ids because they can change upstream due to deformation, but the transform id should match.

There should be validators in place that shape ids should NOT change when publishing animation/pointcache if the shape id was different than on the 'intermediateMesh' in its history. See Validate Animation Out Set Node Ids.

I think we should either refactor it so that we NEVER rely on shape ids to begin with (if we feel that's problematically unsafe) or ALWAYS rely on it if we're doing it in other areas. OR we should document clearly why this particular case is a special case which makes the shape ids extra unsafe than e.g. when they are used for look assignments (which also rely on shape ids for the shader assignment by the way.)

Reply:
Not sure what you want out of this. I need the name of the mesh transform from the container to compare but also to plug into the blendshape. How we archieve this, does not matter to me so up for any suggestions.

Reply:
If this works for the use case it's fine. It just felt odd to me that somehow we're linking meshes but relying on transforms.

Anyway, would it be nicer to use `lib.get_id(transform)` just so we use the implementation that we might want to change up later?

Reply:
Cool, have also documented why we are doing this as well.